source/core/Variants.cpp

```diff
@@ -75,23 +76,32 @@ VIREO_FUNCTION_SIGNATURET(VariantToData, VariantToDataParamBlock)
         TypeRef inputType = _ParamImmediate(InputData._paramType);
         void* inputData = _ParamImmediate(InputData)._pData;
 
-        TypeRef destType = _ParamImmediate(DestData._paramType);
+        TypeRef destType = _ParamImmediate(DestType._paramType);
+
```

Comment:
You now need to verify that destData's actual type is compatible with the wired destType.
The checks below compare variantInnerType and destType, which no longer is guaranteed to be the same as destData's actual type. A mismatch would probably crash.

Reply:
I went ahead and made these changes, though we're protected by semantic analysis when coming from LabVIEW. I am yet to add tests, so hold off on reviewing my recent push. I will resolve this comment when I am done.